src/client/clientmap.cpp

```diff
@@ -34,33 +34,55 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #include <queue>
 
-// struct MeshBufListList
-void MeshBufListList::clear()
+//FIXME: where should I move this to?
+// std::hash for integral types, including ptrs, is identity, which is bad for
+// aligned ptrs.
```

Comment:
do STL types only use the bottom-most bits of a hash or why?

Reply:
Looked into gcc's libstdc++ and llvm's libc++, to be sure, they both do `bucket_index = hashcode % bucket_count` (pseudocode).
This means that if the bucket count is a power of 2 (libc++'s `__constrain_hash` has a special case for power of 2, so I assume it's likely, but not required), only each 16th hash bucket will be used.

Reply:
I see, can you add a short comment explaining this?

Reply:
To be actually sure I'm not writing down bullshit, I've printed out the bucket counts:
13 29 59 127 257  541 1109 2357
It's a bunch of primes, on gcc at least.
This of course makes sense if you consider they want to have a good hash table for simple hash functions.

Sorry, I've talked bullshit!

(I also never noticed any performance difference using my PtrHash btw.. I was just guessing it was because there were not *that* many elements, and so wanted to keep it for extreme cases.)

Considering this, I've removed the PtrHash template now.
If someone finds out that other std lib hashtables behave differently, we can still do this afterwards.